src/app/(user-access)/signup/page.tsx

```diff
@@ -1,3 +1,169 @@
-export default function Page() {
-  return <div></div>;
+'use client';
+
+import { useForm } from 'react-hook-form';
+import { useRouter } from 'next/navigation';
+import Button from '@/components/Button';
+import styles from './signPage.module.css';
+import axiosInstance from '@/lib/axiosInstance';
+import { ERROR_MESSAGES } from '@/constants/message';
+import ModalContainer from '../components/modal/ModalContainer';
+import useModalStore from '../modalStore/modalStore';
+
+type SignupFormInputs = {
+  email: string;
+  nickname: string;
+  password: string;
+  confirmPassword: string;
+  termsAccepted: boolean;
+};
+
+export default function SignupPage() {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid },
+    watch,
+  } = useForm<SignupFormInputs>({ mode: 'onChange' });
+  const router = useRouter();
+
+  const watchPassword = watch('password');
+  const { openModal } = useModalStore();
+
+  const onSubmit = async (data: SignupFormInputs) => {
+    try {
+      await axiosInstance.post('/users', {
+        email: data.email,
+        nickname: data.nickname,
+        password: data.password,
+      });
+      const successMessage = '가입이 완료되었습니다!';
+      openModal(successMessage, 'success');
```

Comment:
변수 따로 선언안해주고 인라인으로 바로 써도 좋을것같네요!
```suggestion
      openModal( '가입이 완료되었습니다!', 'success');
```